src/Microsoft.VisualStudio.Threading/AsyncAutoResetEvent.cs

```diff
@@ -99,7 +99,15 @@ public Task WaitAsync(CancellationToken cancellationToken)
                 else
                 {
                     var waiter = new WaiterCompletionSource(this, cancellationToken, this.allowInliningAwaiters);
-                    this.signalAwaiters.Enqueue(waiter);
+                    if (cancellationToken.IsCancellationRequested)
```

Comment:
There is already ```cancellationToken.IsCancellationRequested``` check at the beginning of the method. There is nothing long-running in between the checks. The second check 99% of the time will be false. So, do we need it at all?